components/agenda/src/Agenda.svelte

```diff
@@ -330,19 +327,22 @@
 
   // Try getting events from 3 sources: first, directly passed in, then, from our store; finally, by way of a fetch
   let calendarEvents: Event[] = [];
-  $: {
-    (async () => {
-      if (events) {
-        calendarEvents = events as Event[];
-      } else {
-        calendarEvents = (await EventStore.getEvents(query)) || [];
-        if (siblingQueries.length) {
-          siblingQueries.forEach((siblingQuery) =>
-            EventStore.getEvents(siblingQuery),
-          );
+  $: setCalendarEvents(), events, query;
+  async function setCalendarEvents() {
+    if (events && Array.isArray(events)) {
+      calendarEvents = events as Event[];
+    } else {
+      await EventStore.getEvents(query).then((nylasEvents) => {
+        if (!events) {
+          calendarEvents = nylasEvents || [];
```

Comment:
now avoids bug where promise takes long and events is set in the meantime

Reply:
Would this not prevent things from updating where the custom data is updated (and passed in again) from the surrounded app? (for example, toggling a different user to see their agenda)?

Reply:
This actually enables custom data to be set over and over again (toggling). When you give new data to `events` it will run to set that data, when you remove that data, it will run to set default nylas data back.